packages/wdio-utils/src/utils.ts

```diff
@@ -7,7 +7,7 @@ import type { Services, Clients } from '@wdio/types'
 
 const SCREENSHOT_REPLACEMENT = '"<Screenshot[base64]>"'
 const SCRIPT_PLACEHOLDER = '"<Script[base64]>"'
-const REGEX_SCRIPT_NAME = /return \(function (\w+)/
+const REGEX_SCRIPT_NAME = /return \((async )*function (\w+)/
```

Comment:
If the * is used to say "none or more" we should probably use ? instead to say "none of one" for a more accurate and performant regex